src/data/nat/factorization.lean

```diff
@@ -159,4 +159,19 @@ end
 @[simp] lemma factorization_prod_pow_eq_self {n : ℕ} (hn : n ≠ 0) : n.factorization.prod pow = n :=
 by simpa only using (multiplicative_factorization id (by simp) (by simp) hn).symm
 
+/-! ### Factorization and divisibility -/
+
+lemma factorization_le_iff_dvd {d n : ℕ} (hd : d ≠ 0) (hn : n ≠ 0) :
+  d.factorization ≤ n.factorization ↔ d ∣ n :=
+begin
+  split,
+  { intro hdn,
+    set K := n.factorization - d.factorization with hK,
+    -- rw dvd_iff_exists_eq_mul_left,
+    use K.prod pow,
+    rw [←(factorization_prod_pow_eq_self hn), ←(factorization_prod_pow_eq_self hd),
```

Comment:
I think these brackets are unnecessary

Reply:
Ah, sorry, I added `add_tsub_cancel_of_le hdn` to the `rw` block rather than pasting your whole suggestion, to make sure I understood what what changing.

Reply:
No worries :)